### PR TITLE
Resilient Thunder Connection

### PR DESCRIPTION
**Pull request by Vinodsathyaseelan**
## What
When Thunder crashed and/or re-started 
1. Recreate Ripple ThunderPool Client and re-connect to Thunder 
2. Reset plugin activation state and re-activate various plugins based on the boot up configuration 
3. Re-subscribe to Thunder events.  

Added unit/integration test to device.thunder_ripple_sdk.src.client
## Why

This changes are needed to restore Thunder session when Thunder process re-starts.

## How

How do these changes achieve the goal?

## Test

How has this been tested? How can a reviewer test it?

## Checklist

- [x] I have self-reviewed this PR
- [ ] I have added tests that prove the feature works or the fix is effective

### CODE REVIEW

File: device/thunder_ripple_sdk/src/client/thunder_plugin.rs [hunk 1]
```diff
@@ -49,6 +50,7 @@ const WIFI_CFG: Cfg = Cfg::new("org.rdk.Wifi", false, false);
 const LOCATION_SYNC: Cfg = Cfg::new("LocationSync", false, false);
 const TTS_CFG: Cfg = Cfg::new("org.rdk.TextToSpeech", false, true);
 const TELEMETRY_CFG: Cfg = Cfg::new("org.rdk.Telemetry", true, false);
+const AUTH_SERVICE_CFG: Cfg = Cfg::new("org.rdk.AuthService", true, false);
```
Comment (satlead) [inline device/thunder_ripple_sdk/src/client/thunder_plugin.rs:53]: Remove this

**Review by satlead — CHANGES_REQUESTED** [review]

File: device/thunder_ripple_sdk/src/client/thunder_plugin.rs [hunk 2]
```diff
@@ -68,6 +70,7 @@ impl ThunderPlugin {
             LocationSync => LOCATION_SYNC,
             TextToSpeech => TTS_CFG,
             Telemetry => TELEMETRY_CFG,
+            AuthService => AUTH_SERVICE_CFG,
```
Comment (satlead) [inline device/thunder_ripple_sdk/src/client/thunder_plugin.rs:73]: Remove this

File: device/thunder_ripple_sdk/src/client/thunder_plugin.rs (hunk 1 above)
Comment (Vinodsathyaseelan) [reply device/thunder_ripple_sdk/src/client/thunder_plugin.rs]: Done.

File: device/thunder_ripple_sdk/Cargo.toml [hunk 3]
```diff
@@ -51,3 +51,4 @@ csv = "=1.1.5"
 base64 = "0.13.0"
 home = { version = "=0.5.5", optional = true}
 regex = "1.7.3"
+tokio-tungstenite = { version = "0.17.1", features = ["native-tls"] }
```
Comment (satlead) [inline device/thunder_ripple_sdk/Cargo.toml:54]: Can we make it a build or test dependency this will add it by default

**Review by tfarz123 — APPROVED** [review]

**Pull request merged**